homeassistant/components/vacuum/__init__.py

```diff
@@ -84,20 +90,38 @@ class VacuumEntityFeature(IntFlag):
 
 # These SUPPORT_* constants are deprecated as of Home Assistant 2022.5.
 # Please use the VacuumEntityFeature enum instead.
-SUPPORT_TURN_ON = 1
-SUPPORT_TURN_OFF = 2
-SUPPORT_PAUSE = 4
-SUPPORT_STOP = 8
-SUPPORT_RETURN_HOME = 16
-SUPPORT_FAN_SPEED = 32
-SUPPORT_BATTERY = 64
-SUPPORT_STATUS = 128
-SUPPORT_SEND_COMMAND = 256
-SUPPORT_LOCATE = 512
-SUPPORT_CLEAN_SPOT = 1024
-SUPPORT_MAP = 2048
-SUPPORT_STATE = 4096
-SUPPORT_START = 8192
+_DEPRECATED_SUPPORT_TURN_ON = DeprecatedConstantEnum(
+    VacuumEntityFeature.TURN_ON, "2025.10"
+)
+_DEPRECATED_SUPPORT_TURN_OFF = DeprecatedConstantEnum(
+    VacuumEntityFeature.TURN_OFF, "2025.10"
+)
+_DEPRECATED_SUPPORT_PAUSE = DeprecatedConstantEnum(VacuumEntityFeature.PAUSE, "2025.10")
+_DEPRECATED_SUPPORT_STOP = DeprecatedConstantEnum(VacuumEntityFeature.STOP, "2025.10")
+_DEPRECATED_SUPPORT_RETURN_HOME = DeprecatedConstantEnum(
+    VacuumEntityFeature.RETURN_HOME, "2025.10"
+)
+_DEPRECATED_SUPPORT_FAN_SPEED = DeprecatedConstantEnum(
+    VacuumEntityFeature.FAN_SPEED, "2025.10"
+)
+_DEPRECATED_SUPPORT_BATTERY = DeprecatedConstantEnum(
+    VacuumEntityFeature.BATTERY, "2025.10"
+)
+_DEPRECATED_SUPPORT_STATUS = DeprecatedConstantEnum(
+    VacuumEntityFeature.STATUS, "2025.10"
+)
+_DEPRECATED_SUPPORT_SEND_COMMAND = DeprecatedConstantEnum(
+    VacuumEntityFeature.SEND_COMMAND, "2025.10"
+)
+_DEPRECATED_SUPPORT_LOCATE = DeprecatedConstantEnum(
+    VacuumEntityFeature.LOCATE, "2025.10"
+)
+_DEPRECATED_SUPPORT_CLEAN_SPOT = DeprecatedConstantEnum(
+    VacuumEntityFeature.CLEAN_SPOT, "2025.10"
+)
+_DEPRECATED_SUPPORT_MAP = DeprecatedConstantEnum(VacuumEntityFeature.MAP, "2025.10")
+_DEPRECATED_SUPPORT_STATE = DeprecatedConstantEnum(VacuumEntityFeature.STATE, "2025.10")
```

Comment:
Side note: We should investigate if we can remove this supported feature now since there's only one vacuum entity type remaining. Is the frontend still using this supported feature to identify the entity type?

Reply:
I will look and see if it's used (and if so, how).